astroid/nodes/node_ng.py

```diff
@@ -509,7 +510,7 @@ def as_string(self):
         :returns: The source code.
         :rtype: str
         """
-        return as_string.to_code(self)
+        return AsStringVisitor("    ")(self)
```

Comment:
Maybe we could add 4 spaces as default for `AsStringVisitor.indent` in `__init__`?

Reply:
This comment is still open

Reply:
Done.